helios-client/src/main/java/com/spotify/helios/common/descriptors/Job.java

```diff
@@ -633,6 +654,16 @@ public Builder setNetworkMode(final String networkMode) {
       return this;
     }
 
+    public Builder setMetadata(final Map<String, String> metadata) {
+      p.metadata = metadata;
+      return this;
+    }
+
+    public Builder addMetadata(String name, String value) {
```

Comment:
Calling `addMetadata()` will throw an exception unless you called `setMetadata()` (with a mutable map) first, because `p.metadata` defaults to `emptyMap()` which is immutable.

`metadata` in the builder should probably default to `Maps.newHashMap()`.
`setMetadata()` should make a copy of the input map.